tests/test_cubicspline.py

```diff
@@ -0,0 +1,121 @@
+    with pytest.raises(cubicspline.InterpolationException):
+        cubicspline.Interpolater(x, y)
+
+    # invalid values
+    y[1] = np.inf
+    with pytest.raises(cubicspline.InterpolationException):
+        cubicspline.Interpolater(x, y)
+    y[1] = np.nan
+    with pytest.raises(cubicspline.InterpolationException):
+        cubicspline.Interpolater(x, y)
+
+
+def relative_error(test_value, true_value):
+    return abs((test_value - true_value) / true_value)
+
+
+@pytest.mark.parametrize(
+    "interpolater", [(cubicspline.Interpolater), (cubicspline.LogInterpolater)]
+)
+def test_constant(interpolater):
+    """Interpolate and extrapolate f(x) = 1."""
+    x = np.arange(1, 8)
+    y = np.ones(7)
+    data = np.dstack((x, y))[0]
+    p = interpolater(data)
+
+    # two different ways to do the same thing:
+    assert (p(np.asarray([0.025, 1, 2.5, 4, 5.55, 7.01, 19])) == 1).all()
+    if interpolater == cubicspline.LogInterpolater:
+        assert (
+            p.value_log_array(np.asarray([0.0125, 1, 2.5, 4, 5.55, 7.01, 19])) == 1
+        ).all()
+    else:
+        assert (
+            p.value_array(np.asarray([-0.025, 1, 2.5, 4, 5.55, 7.01, 19])) == 1
+        ).all()
+
+
+@pytest.mark.parametrize(
+    "interpolater", [(cubicspline.Interpolater), (cubicspline.LogInterpolater)]
+)
+def test_linear(interpolater):
+    """Interpolate f(x) = 10x."""
+    x = np.arange(1, 5)
+    y = np.asarray([10, 20, 30, 40])
+    data = np.dstack((x, y))[0]
+    p = interpolater(data)
+
+    if interpolater == cubicspline.Interpolater:
+        assert p(-1) == -10
+        assert p(0) == 0
+    assert pytest.approx(p.value(0.5), 5)
+    assert pytest.approx(p(1) == 10)
+    assert pytest.approx(p(1.75) == 17.5)
+    assert pytest.approx(p(2) == 20)
+    assert pytest.approx(p(2.2) == 22)
+    assert pytest.approx(p(3) == 30)
+    assert pytest.approx(p(4) == 40)
+
+
+@pytest.mark.parametrize(
+    "interpolater", [(cubicspline.Interpolater), (cubicspline.LogInterpolater)]
+)
+def test_random(interpolater):
+    """Interpolate random points."""
+    x = np.arange(1, 5)
+    y = np.asarray([1.67, 1.99, 0.465, 0.234])
+    data = np.dstack((x, y))[0]
+    p = interpolater(data)
+
+    for i, x_ in enumerate(x):
+        assert p(x_) == y[i]
+
+
+def test_polynomial():
+    """Interpolate f(x) = 1 + 2x + 3x**2."""
+    f = np.polynomial.polynomial.Polynomial((1, 2, 3))
+    x = np.arange(0, 1000, 0.01)
+    y = np.array([f(i) for i in x])
+    interpolater = cubicspline.Interpolater(x, y)
+
+    for x_ in np.asarray([0, 1, 0.0998, 456, 666.666, 998.501, 999.98, 99.98999]):
+        assert relative_error(interpolater(x_), f(x_)) < 1e-7
+    t = interpolater.test(0, 10, 100)
+    for x_, y_ in t:
+        assert relative_error(y_, f(x_)) < 1e-7
+
+
+@pytest.mark.xfail
+def test_polynomial_edge():
+    """
+    Interpolate f(x) = 1 + 2x + 3x**2.
+
+    Tests values close to the lower edge of the interpolation function.
+
+    This test currently fails.
```

Comment:
Fails on James' branch? Or is this test not part of the build process, and that is why the build is green.

Reply:
Should we open an issue to investigate the test failure?